src/libs/actions/Report.js

```diff
@@ -495,6 +495,84 @@ function fetchChatReports() {
         });
 }
 
+/**
+ * Get a simplified version of an IOU report
+ *
+ * @param {Object} reportData
+ * @param {Number} reportData.transactionID
+ * @param {Number} reportData.amount
+ * @param {String} reportData.currency
+ * @param {String} reportData.created
+ * @param {String} reportData.comment
+ * @param {Object[]} reportData.transactionList
+ * @param {String} reportData.ownerEmail
+ * @param {String} reportData.managerEmail
+ * @param {Number} reportData.reportID
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportData) {
+    const transactions = _.map(reportData.transactionList, transaction => ({
+        transactionID: transaction.transactionID,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        created: transaction.created,
+        comment: transaction.comment,
+    }));
+
+    return {
+        reportID: reportData.reportID,
+        ownerEmail: reportData.ownerEmail,
+        managerEmail: reportData.managerEmail,
+        currency: reportData.currency,
+        transactions,
+    };
+}
+
+/**
+ * Fetches the updated data for an IOU Report and updates the IOU collection in Onyx
+ *
+ * @param {Object[]} reportHistory
+ */
+function updateIOUReportData(reportHistory) {
+    const containsIOUAction = _.any(reportHistory, action => action.actionName === 'IOU');
+
+    // If there aren't any IOU actions, we don't need to fetch any additional data
+    if (!containsIOUAction) {
+        return;
+    }
+
+    const otherParticipants = _.chain(reportHistory)
+        .pluck('actorEmail')
+        .unique()
+        .without(currentUserEmail)
+        .value();
```

Comment:
@Julesssss my guess is this is where the `undefined` participant is coming from. I think there needs to be a `.flatten()` added into the end of this chain to remove it (this is pretty typical).